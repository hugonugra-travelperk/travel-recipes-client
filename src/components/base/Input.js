import styled from "styled-components";

export const InputForm = styled.div`
  display: flex;
  align-items: center;
`

export const Input = styled.input`
  font-size: 18px;
  padding: 10px;
  margin: 10px;
  background: papayawhip;
  border: none;
  border-radius: 3px;
  ::placeholder {
    color: palevioletred;
  }
`;

export const InputArea = styled.textarea` 
  font-size: 18px;
  padding: 10px;
  margin: 10px;
  background: papayawhip;
  border: none;
  border-radius: 3px;
  ::placeholder {
    color: palevioletred;
  }
`;
