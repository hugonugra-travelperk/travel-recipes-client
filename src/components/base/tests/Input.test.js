import {render, cleanup} from "@testing-library/react"
import "@testing-library/jest-dom";
import {InputForm, Input, InputArea} from "../Input"

it("renders", () => {
  const {asFragment} = render(<InputForm>Test text</InputForm>);
  expect(asFragment()).toMatchSnapshot();
});

it("renders", () => {
  const {asFragment} =
    render(
      <Input
        placeholder={'test text'}
        defaultValue={''}
      />
    );
  expect(asFragment()).toMatchSnapshot();
});

it("renders", () => {
  const {asFragment} =
    render(
      <InputArea
        placeholder={'test text'}
        defaultValue={''}
      />
    );
  expect(asFragment()).toMatchSnapshot();
});
