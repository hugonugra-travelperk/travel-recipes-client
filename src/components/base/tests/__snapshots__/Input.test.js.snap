// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL gowonf"
  >
    Test text
  </div>
</DocumentFragment>
`;

exports[`renders 2`] = `
<DocumentFragment>
  <input
    class="sc-gsDKAQ jJyxYW"
    placeholder="test text"
    value=""
  />
</DocumentFragment>
`;

exports[`renders 3`] = `
<DocumentFragment>
  <textarea
    class="sc-dkPtRN dMlLcC"
    placeholder="test text"
  />
</DocumentFragment>
`;
